.github/workflows/integration.yml

```diff
@@ -0,0 +1,48 @@
+name: integration
+
+on:
+  push:
+    paths-ignore:
+      - 'docs/**'
+      - '*.md'
+  pull_request:
+    paths-ignore:
+      - 'docs/**'
+      - '*.md'
+
+jobs:
+  install-production:
+    runs-on: ${{ matrix.os }}
+    permissions:
+      contents: read
+    
+    strategy:
+      matrix:
+        node-version: [14, 16, 18]
+        os: [ubuntu-latest]
+
+    steps:
+      - uses: actions/checkout@v3
```

Comment:
```suggestion
      - uses: actions/checkout@main
```

Reply:
why main?

Reply:
the version of the checkout action doesn't have anything to do with node

Reply:
Yes, it does -- https://github.com/actions/checkout/blob/2541b1294d2704b0964813337f33b291d3f8596b/README.md?plain=1#L17

Reply:
why do we care about the version of node that the action uses to run? it's only checking out the code, as far as we're concerned it could be running php and it wouldn't make any difference to us or anybody else using the action. also note this is inconsistent with the way actions are being used pretty much anywhere else, and I don't think it's recommended to use branches as references for actions.

Reply:
It matters because we have to update the workflows. I am trying to get all workflows standardizing on using `@main` from the ***actions/**** suite of actions. These are maintained by GitHub. Following `@main` on those is as good as following a tag. For third party actions, i.e. those _not_ maintained by GitHub, we should be using a direct SHA reference.

Reply:
I do not agree for a couple of reasons:

- using a branch reference you're using something unpublished which, as far as we know, could very well be broken
- dependabot can and does automate bumping of action versions
- referencing a tag, which is the way [github recommends using actions](https://docs.github.com/en/actions/creating-actions/about-custom-actions#using-tags-for-release-management), gives you the latest version within that tag

Reply:
Feel free to open an issue for general discussion.

Reply:
To be honest I believe you should have opened an issue for general discussion because what you suggested to do here deviates from what's already being done in this and all other repos. I don't personally mind, but you have introduced the inconsistency by suggesting this change.